lib/data/utils.js

```diff
@@ -0,0 +1,53 @@
+/**
+ * proxyCompareURL - compares request endpoint to urls in NO_PROXY env var
+ * @param {string} endpoint - url of request
+ * @return {bool} true if request endpoint matches no proxy, false if not
+ */
+function proxyCompareURL(endpoint) {
+    const noProxy = process.env.NO_PROXY || process.env.no_proxy;
+    if (!noProxy) {
+        return false;
+    }
+    // noProxy env var is a comma separated list of urls not to proxy
+    const noProxyList = noProxy.split(',');
+    if (noProxyList.includes(endpoint)) {
+        return true;
+    }
+    const epArr = endpoint.split('.');
+    // reverse array to make comparison easier
+    epArr.reverse();
+    let match = false;
+    for (let j = 0; j < noProxyList.length; j++) {
```

Comment:
You may want to look at this for later refinement: https://github.com/isaacs/minimatch